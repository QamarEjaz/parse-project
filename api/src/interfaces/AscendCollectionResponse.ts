import { AscendMetaData } from "./AscendMetaData";

export interface AscendCollectionResponse<T> {
    statusCode: number;
    data: T[];
    meta: AscendMetaData;
}

