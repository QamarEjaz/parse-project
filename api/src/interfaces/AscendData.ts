export interface AscendData {
    id: string;
    [key: string]: any;
}