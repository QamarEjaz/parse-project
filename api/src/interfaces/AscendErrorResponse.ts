export interface AscendErrorData {
    code: string;
    title: string;
    description: string;
    source: string;
}

export interface AscendErrorResponse {
    statusCode: number;
    errors: AscendErrorData[];
}