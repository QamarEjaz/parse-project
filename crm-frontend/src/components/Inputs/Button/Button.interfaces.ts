export interface IButtonProps {
  type?: "button" | "submit" | "reset" | undefined
  rounded?: boolean
  className?: string
  loading?: boolean
  variant?: "text" | "contained" | "outlined"
  color?:
    | "primary"
    | "indigo"
    | "indigo-light"
    | "red"
    | "gray"
    | "gray-light"
    | "gray-dark"
    | "white"
  disabled?: boolean
  onClick?: () => void
  style?: {}
  children?: React.ReactNode
}
