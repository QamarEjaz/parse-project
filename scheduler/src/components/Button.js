import Spinner from "./Spinner";

const variants = {
  primary: "text-white bg-gray-800 hover:bg-black",
  gray: "border border-gray-300 hover:bg-gray-50 text-gray-600",
  danger: "text-white bg-red-400 hover:bg-red-500",
};

const Button = ({
  title = "",
  className = "px-10 py-2",
  variant = "primary",
  loading = false,
  children,
  type = "button",
  disabled,
  onClick = () => {},
}) => {
  return (
    <button
      type={type}
      className={`flex shadow-sm rounded-md uppercase transition-colors ${className} ${variants[variant]}`}
      onClick={onClick}
      disabled={disabled}
    >
      <span>{title}</span>

      {children}

      {loading && (
        <span className="ml-3">
          <Spinner />
        </span>
      )}
    </button>
  );
};

export default Button;
