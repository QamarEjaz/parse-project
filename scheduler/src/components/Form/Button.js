import { useMemo } from "react";

const Button = ({
  type,
  className,
  children,
  loading,
  variant = "contained",
  color = "indigo",
  disabled,
  onClick,
  style,
}) => {
  const variantClasses = useMemo(
    () => ({
      text: "btn-text",
      contained: "btn-contained",
      outlined: "btn-outlined",
    }),
    []
  );

  const colorClasses = useMemo(
    () => ({
      primary: "btn-primary",
      indigo: "btn-indigo",
      "indigo-light": "btn-indigo-light",
      red: "btn-red",
      gray: "btn-gray",
    }),
    []
  );

  return (
    <button
      type={type || "button"}
      className={`btn ${variantClasses[variant]} ${colorClasses[color]} ${className}`}
      disabled={disabled || loading}
      onClick={onClick}
      style={style}
    >
      {loading && (
        <div className="absolute top-1/2 left-1/2 transform -translate-x-1/2 -translate-y-1/2">
          <div className="spinner-wrapper">
            <Spinner
              className={
                !color || color === "gray" || color === "white"
                  ? "bg-gray-700"
                  : variant !== "contained"
                  ? color === "primary"
                    ? "bg-primary"
                    : color === "indigo" || color === "indigo-light"
                    ? "bg-indigo-500"
                    : color === "red"
                    ? "bg-red-500"
                    : "bg-gray-700"
                  : "bg-white"
              }
            />
          </div>
        </div>
      )}
      <span className={`${loading && "opacity-0"}`}>{children}</span>
    </button>
  );
};

export default Button;

const Spinner = ({ className }) => {
  return (
    <div className={`lds-ellipsis`}>
      <div className={className || ""}></div>
      <div className={className || ""}></div>
      <div className={className || ""}></div>
      <div className={className || ""}></div>
    </div>
  );
};
