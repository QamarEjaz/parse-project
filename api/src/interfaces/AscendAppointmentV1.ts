import { AscendData } from "./AscendData";
import { AscendReference } from "./AscendReference";

export interface AscendAppointmentUpdateV1 {
    confirmed?: string;
    needsFollowUp?: boolean;
    followedUpOn?: string;
    bookingType?: string;
    needsPremedicate?: boolean;
    note?: string;
    other?: string;
    bookedOnline?: boolean;
    leftMessage?: string;
    duration?: number;
    lastModified?: string;
    labCaseDentalLab?: AscendReference;
    labCaseStatus?: string;
    labCaseDueDate?: string;
    labCaseNote?: string;
    patientProcedures?: AscendReference[];
    practiceProcedures?: AscendReference[];
    visits?: AscendReference[];
    location?: AscendReference;
}

export interface AscendAppointmentV1 extends AscendData, AscendAppointmentUpdateV1 {
    title?: string;
    start: string;
    end?: string;
    created?: string;
    duration?: number;
    status: string;
    lastModified?: string;
    labCaseDentalLab?: AscendReference;
    labCaseStatus?: string;
    labCaseDueDate?: string;
    labCaseNote?: string;
    provider: AscendReference;
    otherProvider?: AscendReference;
    patient: AscendReference;
    operatory: AscendReference;

}