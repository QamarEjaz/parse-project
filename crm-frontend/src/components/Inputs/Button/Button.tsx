import { useMemo } from "react"
import { IButtonProps } from "./Button.interfaces"

const variantClasses = {
  text: "btn-text",
  contained: "btn-contained",
  outlined: "btn-outlined",
}

const colorClasses = {
  primary: "btn-primary",
  indigo: "btn-indigo",
  "indigo-light": "btn-indigo-light",
  red: "btn-red",
  gray: "btn-gray",
  "gray-light": "btn-gray-light",
  "gray-dark": "btn-gray-dark",
}

const Button = ({ rounded, type = "button", className, children, loading, variant = "contained", color, disabled, onClick, style }: IButtonProps): JSX.Element => {
  const loader = useMemo(
    () =>
      Array(4)
        .fill(null)
        .map((_, index) => (
          <div
            key={index}
            className={
              !color || (color === "gray" && variant !== "contained") || color === "white" ? "bg-gray-700" : variant !== "contained" ? (color === "primary" ? "bg-primary" : color === "indigo" || color === "indigo-light" ? "bg-indigo-500" : color === "red" ? "bg-red-500" : "bg-gray-700") : "bg-white"
            }
          />
        )),
    [color, variant]
  )

  return (
    <button id="button" type={type} className={`btn ${rounded ? "btn-rounded" : ""} ${variantClasses[variant as keyof typeof variantClasses]} ${colorClasses[color as keyof typeof colorClasses]} ${className}`} disabled={disabled || loading} onClick={onClick} style={style}>
      {loading ? (
        <div className="absolute top-1/2 left-1/2 transform -translate-x-1/2 -translate-y-1/2" id="button-loader">
          <div className="spinner-wrapper">
            <div className="lds-ellipsis">{loader}</div>
          </div>
        </div>
      ) : null}
      <div className={`${loading && "opacity-0"}`} id="button-text">
        {children}
      </div>
    </button>
  )
}

export default Button
