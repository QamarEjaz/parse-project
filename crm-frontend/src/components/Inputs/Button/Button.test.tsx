import { shallow } from "enzyme"
import Button from "./Button"
import { IButtonProps } from "./Button.interfaces"

describe("<Button> Component", () => {
  it("expect to render Button component without props", () => {
    expect(shallow(<Button />)).toMatchSnapshot()
  })
  it("expect to render Button component with props", () => {
    const mockProps: IButtonProps = { children: "Click me!", variant: "contained", color: "gray-dark" }
    expect(shallow(<Button {...mockProps} />)).toMatchSnapshot()
  })
  it("expect to render Button with text of Hello World", () => {
    const mockProps: IButtonProps = { children: "Hello World!", variant: "contained", color: "gray-dark" }
    const wrapper = shallow(<Button {...mockProps} />)

    expect(wrapper.find("#button-text").text()).toContain("Hello World!")
  })
  it("expect Button click event to work", () => {
    let isClickWorking = false

    const mockProps: IButtonProps = { children: "Click Me!", variant: "contained", color: "gray-dark", onClick: () => (isClickWorking = true) }

    const wrapper = shallow(<Button {...mockProps} />)

    wrapper.find("#button").simulate("click")

    expect(isClickWorking).toEqual(true)
  })
  it("expect Button loading to work", () => {
    const mockProps: IButtonProps = { children: "Click Me!", variant: "contained", color: "gray-dark", loading: true }

    const wrapper = shallow(<Button {...mockProps} />)

    expect(wrapper.find("#button-loader").exists()).toEqual(true)
  })
})
