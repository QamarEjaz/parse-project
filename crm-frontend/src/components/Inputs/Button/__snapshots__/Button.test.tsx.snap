// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> Component expect to render Button component with props 1`] = `
<button
  className="btn  btn-contained btn-gray-dark undefined"
  id="button"
  type="button"
>
  <div
    className="undefined"
    id="button-text"
  >
    Click me!
  </div>
</button>
`;

exports[`<Button> Component expect to render Button component without props 1`] = `
<button
  className="btn  btn-contained undefined undefined"
  id="button"
  type="button"
>
  <div
    className="undefined"
    id="button-text"
  />
</button>
`;
